Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,288 @@
+    inv_mul_eq_div, div_self (ne_of_gt ha), div_one, Function.comp.assoc] at this
+  convert this
+  ext
+  simp [inv_mul_eq_div, div_self (ne_of_gt ha)]
+
+lemma IsIntegralCurveAt.comp_neg {γ : ℝ → M} (hγ : IsIntegralCurveAt γ v t₀ x₀) :
+    IsIntegralCurveAt (γ ∘ Neg.neg) (-v) (-t₀) x₀ := by
+  obtain ⟨h1, ε, hε, h2⟩ := hγ
+  refine' ⟨by simp [h1], ε, hε, _⟩
+  intros t ht
+  rw [←neg_add', neg_add_eq_sub, ←neg_sub, ←neg_mem_Ioo_iff] at ht
+  have h2' := h2 (-t) ht
+  rw [Function.comp_apply, Pi.neg_apply, ←neg_one_smul ℝ (v (γ (-t))),
+    ←ContinuousLinearMap.smulRight_comp]
+  apply HasMFDerivAt.comp t h2'
+  refine' ⟨continuousAt_neg, _⟩
+  simp only [writtenInExtChartAt, extChartAt, LocalHomeomorph.extend,
+    LocalHomeomorph.refl_localEquiv, LocalEquiv.refl_source,
+    LocalHomeomorph.singletonChartedSpace_chartAt_eq, modelWithCornersSelf_localEquiv,
+    LocalEquiv.trans_refl, LocalEquiv.refl_coe, LocalEquiv.refl_symm, Function.comp.right_id,
+    Function.comp.left_id, modelWithCornersSelf_coe, range_id, id_eq, hasFDerivWithinAt_univ]
+  apply HasDerivAt.hasFDerivAt
+  exact hasDerivAt_neg _
+
+lemma isIntegralCurveAt_comp_neg {γ : ℝ → M} :
+    IsIntegralCurveAt γ v t₀ x₀ ↔ IsIntegralCurveAt (γ ∘ Neg.neg) (-v) (-t₀) x₀ := by
+  refine' ⟨fun hγ => IsIntegralCurveAt.comp_neg hγ, _⟩
+  intro hγ
+  have := hγ.comp_neg
+  rw [Function.comp.assoc, neg_comp_neg, neg_neg, neg_neg] at this
+  exact this
+
+lemma IsIntegralCurveAt.comp_mul_ne_zero {γ : ℝ → M} (hγ : IsIntegralCurveAt γ v t₀ x₀) {a : ℝ}
+    (ha : a ≠ 0) : IsIntegralCurveAt (γ ∘ (fun t => t * a)) (a • v) (t₀ / a) x₀ := by
+  rw [ne_iff_lt_or_gt] at ha
+  cases' ha with ha ha
+  · apply isIntegralCurveAt_comp_neg.mpr
+    have : (fun t ↦ t * a) ∘ Neg.neg = fun t ↦ t * -a := by ext; simp
+    rw [Function.comp.assoc, this, ←neg_smul, ←div_neg]
+    exact hγ.comp_mul_pos (neg_pos_of_neg ha)
+  · exact hγ.comp_mul_pos ha
+
+lemma isIntegralCurveAt_comp_mul_ne_zero {γ : ℝ → M} {a : ℝ} (ha : a ≠ 0) :
+    IsIntegralCurveAt γ v t₀ x₀ ↔ IsIntegralCurveAt (γ ∘ (fun t => t * a)) (a • v) (t₀ / a) x₀ := by
+  refine' ⟨fun hγ => IsIntegralCurveAt.comp_mul_ne_zero hγ ha, _⟩
+  intro hγ
+  have := hγ.comp_mul_ne_zero (inv_ne_zero ha)
+  rw [smul_smul, inv_mul_eq_div, div_self ha, one_smul, ←div_mul_eq_div_div_swap,
+    inv_mul_eq_div, div_self ha, div_one, Function.comp.assoc] at this
+  convert this
+  ext
+  simp [inv_mul_eq_div, div_self ha]
+
+/-- For any continuously differentiable vector field and any chosen non-boundary point `x₀` on the
+  manifold, there exists an integral curve `γ : ℝ → M` such that `γ t₀ = x₀` and the tangent vector
+  of `γ` at `t` coincides with the vector field at `γ t` for all `t` within an open interval around
+  `t₀`.-/
+theorem exists_integralCurve_of_contMDiff_tangent_section (hx : I.IsInteriorPoint x₀) :
+    ∃ (γ : ℝ → M), IsIntegralCurveAt γ v t₀ x₀ := by
+  /- express the derivative of the section `v` in the local charts -/
```

Comment:
I think normal comments `with --` would suffice here. Not sure if that's in the style guide or just my personal opinion :-)

Reply:
Just using `/- ... -/` for multiline comments at the moment. I assume a lot of these comments will be shortened / moved elsewhere once we extract a few lemmas from the proof. Eventually let's use `--`.